src/devices/Sn74hc595/Sn74hc595.cs

```diff
@@ -0,0 +1,361 @@
+        public Sn74hc595(SpiDevice spiDevice, int count = 1)
+        {
+            _spiDevice = spiDevice;
+            _count = count;
+        }
+
+        /// <summary>
+        /// Initialize a new Sn74hc595 device connected through both SPI and GPIO (SPI for writing data; GPIO for configuration; use 4-5 pins)
+        /// </summary>
+        /// <param name="spiDevice">SpiDevice used for serial communication.</param>
+        /// <param name="pinMapping">The pin mapping to use by the binding</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance</param>
+        /// <param name="count">Count of (daisy-chained) shift registers. Default/minimum is 1.</param>
+        public Sn74hc595(SpiDevice spiDevice, PinMapping pinMapping, GpioController gpioController = null, bool shouldDispose = true, int count = 1)
+                  : this(pinMapping, gpioController, shouldDispose, count)
+        {
+            _spiDevice = spiDevice;
+        }
+
+        /// <summary>
+        /// Count of shift units.
+        /// The number of (daisy-chained) shift registers. Minimum is 1.
+        /// Not the count of registers on a single unit.
+        /// </summary>
+        public int Count => _count;
+
+        /// <summary>
+        /// Count of total bits / registers across all (daisy-chained) shift registers.
+        /// Minimum is 8.
+        /// </summary>
+        public int Bits => _count * 8;
+
+        /// <summary>
+        /// Reports if Sn74hc595 is controlled with SPI.
+        /// </summary>
+        public bool UsesSpi => _spiDevice is object;
+
+        /// <summary>
+        /// Reports if Sn74hc595 is controlled with GPIO.
+        /// </summary>
+        public bool UsesGpio => _controller is object;
+
+        /// <summary>
+        /// Clear storage registers.
+        /// </summary>
+        public void ClearStorage()
+        {
+            if (_controller is null || _pinMapping.SrClr == 0)
+            {
+                throw new ArgumentNullException($"{nameof(ClearStorage)}: GpioController was not provided or {nameof(_pinMapping.SrClr)} not mapped to pin");
+            }
+
+            _controller.Write(_pinMapping.SrClr, 0);
+            _controller.Write(_pinMapping.SrClr, 1);
+        }
+
+        /// <summary>
+        /// Shifts zeros.
+        /// Will dim all connected LEDs, for example.
+        /// </summary>
+        public void ShiftClear()
+        {
+            for (int i = 0; i < Count; i++)
+            {
+                ShiftByte(0);
+            }
+        }
+
+        /// <summary>
+        /// Shifts single value to next register
+        /// Does not perform latch.
+        /// </summary>
+        public void Shift(PinValue value)
```

Comment:
do we expect folks to call this directoy or to always call ShiftByte instead?

Reply:
"directory"? Both Shift and ShiftByte are separate use cases. This comment pretty much clarifies that: https://github.com/dotnet/iot/pull/1104#issuecomment-644595161

Reply:
Will make a ShiftBit method. Add in the comment that it is only supported via GPIO.